ipa-core/src/bin/report_collector.rs

```diff
@@ -171,25 +203,150 @@ fn gen_inputs(
+    let path = if Path::is_file(path) {
+        let mut new_file_name = thread_rng()
+            .sample_iter(&Alphanumeric)
+            .take(5)
+            .map(char::from)
+            .collect::<String>();
+        let file_name = path.file_stem().ok_or("not a file")?;
+
+        new_file_name.insert(0, '-');
+        new_file_name.insert_str(0, &file_name.to_string_lossy());
+        tracing::warn!(
+            "{} file exists, renaming to {:?}",
+            path.display(),
+            new_file_name
+        );
+
+        // it will not be 100% accurate until file_prefix API is stabilized
+        Cow::Owned(
+            path.with_file_name(&new_file_name)
+                .with_extension(path.extension().unwrap_or("".as_ref())),
+        )
+    } else {
+        Cow::Borrowed(path)
+    };
+    let mut file = File::options()
+        .write(true)
+        .create_new(true)
+        .open(path.deref())
+        .map_err(|e| format!("Failed to create output file {}: {e}", path.display()))?;
+
+    write!(file, "{}", serde_json::to_string_pretty(query_result)?)?;
+    Ok(())
+}
+
 async fn ipa(
     args: &Args,
-    network: &NetworkConfig,
     security_model: IpaSecurityModel,
     ipa_query_config: IpaQueryConfig,
     helper_clients: &[MpcHelperClient; 3],
     query_style: IpaQueryStyle,
 ) -> Result<(), Box<dyn Error>> {
-    let input = InputSource::from(&args.input);
-    let query_type: QueryType;
-    match (security_model, &query_style) {
-        (IpaSecurityModel::SemiHonest, IpaQueryStyle::Oprf) => {
-            query_type = QueryType::OprfIpaRelaxedDpPadding(ipa_query_config);
+    let query_type = get_query_type(security_model, &query_style, ipa_query_config);
+
+    let enc1 = &args
+        .enc_input_file1
+        .as_ref()
+        .expect("--enc_input_file1 is required");
+    let enc2 = &args
+        .enc_input_file1
+        .as_ref()
+        .expect("--enc_input_file2 is required");
+    let enc3 = &args
+        .enc_input_file1
+        .as_ref()
+        .expect("--enc_input_file3 is required");
+
+    let mut buffers: [_; 3] = std::array::from_fn(|_| Vec::new());
+    for (i, path) in [enc1, enc2, enc3].iter().enumerate() {
+        let file = File::open(path).unwrap_or_else(|e| panic!("unable to open file {path:?}. {e}"));
+        let reader = BufReader::new(file);
+        for line in reader.lines() {
+            let line = line.unwrap();
+            let encrypted_report_bytes = hex::decode(line.trim()).unwrap();
+            buffers[i].put_u16_le(encrypted_report_bytes.len().try_into().unwrap());
+            buffers[i].put_slice(encrypted_report_bytes.as_slice());
         }
-        (IpaSecurityModel::Malicious, IpaQueryStyle::Oprf) => {
-            panic!("OPRF for malicious is not implemented as yet")
+    }
+    // Panic if input sizes are not the same
+    assert!(buffers[0].len() == buffers[1].len());
+    assert!(buffers[1].len() == buffers[2].len());
```

Comment:
```suggestion
    assert_eq!(buffers[0].len(), buffers[1].len());
    assert_eq!(buffers[1].len(), buffers[2].len());
```

Reply:
or you can return an error too - usually it is better to handle user input through result and error propagation rather than panic. Not that it matters here that much, but just for consistency reasons it may be better to do it this way

Reply:
look at that, and generally errors are reserved for something that might be recoverable. this seems unrecoverable. I added a comment to that effect, but lmk if you disagree.